package/kernel/linux/modules/virt.mk

```diff
@@ -72,3 +72,47 @@ define KernelPackage/kvm-amd/description
 endef
 
 $(eval $(call KernelPackage,kvm-amd))
+
+
+define KernelPackage/vfio
+  SUBMENU:=Virtualization
+  TITLE:=VFIO Non-Privileged userspace driver framework
+  DEPENDS:=@TARGET_x86_generic||TARGET_x86_64
+  KCONFIG:= \
+	CONFIG_VFIO \
+	CONFIG_VFIO_NOIOMMU=n \
+	CONFIG_VFIO_MDEV=n
+  FILES:= \
+	$(LINUX_DIR)/drivers/vfio/vfio.ko \
+	$(LINUX_DIR)/drivers/vfio/vfio_virqfd.ko \
+	$(LINUX_DIR)/drivers/vfio/vfio_iommu_type1.ko
+  AUTOLOAD:=$(call AutoProbe,vfio.ko vfio_iommu_type1.ko vfio_virqfd.ko)
+endef
+
+define KernelPackage/vfio/description
+  VFIO provides a framework for secure userspace device drivers.
+endef
+
+$(eval $(call KernelPackage,vfio))
+
+
+define KernelPackage/vfio-pci
+  SUBMENU:=Virtualization
+  TITLE:=Generic VFIO support for any PCI device
+  DEPENDS:=@TARGET_x86_generic||TARGET_x86_64 @PCI_SUPPORT +kmod-vfio +kmod-irqbypass
+  KCONFIG:= \
+	CONFIG_VFIO_PCI \
+	CONFIG_VFIO_PCI_IGD=y
+  FILES:= $(LINUX_DIR)/drivers/vfio/pci/vfio-pci.ko
+  ifneq ($(wildcard $(LINUX_DIR)/drivers/vfio/pci/vfio-pci-core.ko),)
+    FILES+=$(LINUX_DIR)/drivers/vfio/pci/vfio-pci-core.ko
+  endif
+  AUTOLOAD:=$(call AutoProbe,vfio-pci-core.ko vfio-pci.ko)
```

Comment:
I would prefer this:
````
CONFIG_VFIO_PCI_IGD=y
  FILES:= \
	$(LINUX_DIR)/drivers/vfio/pci/vfio-pci.ko \
	$(LINUX_DIR)/drivers/vfio/pci/vfio-pci-core.ko@ge5.15
  AUTOLOAD:=$(call AutoProbe,vfio-pci)
````
This will add the vfio-pci-core.ko module on kernel version >= 5.15. I assume that vfio-pci.ko has a dependency to vfio-pci-core.ko in case it is needed, then it should also get loaded automatically.